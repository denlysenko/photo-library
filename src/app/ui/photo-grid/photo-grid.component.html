<div class="photo-grid">
  <app-photo-tile *ngFor="let photo of photos; trackBy: trackByFn"
                  [photo]="photo"
                  (click)="photoClicked.emit(photo)"
                  data-testid="photo-tile"></app-photo-tile>
</div>