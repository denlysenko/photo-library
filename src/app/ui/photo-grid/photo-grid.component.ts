import {
  ChangeDetectionStrategy,
  Component,
  EventEmitter,
  Input,
  Output,
} from '@angular/core';

import { Photo } from '../../interfaces';

@Component({
  selector: 'app-photo-grid',
  templateUrl: './photo-grid.component.html',
  styleUrls: ['./photo-grid.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class PhotoGridComponent {
  @Input() photos: Photo[];
  @Output() photoClicked = new EventEmitter<string>();
}
