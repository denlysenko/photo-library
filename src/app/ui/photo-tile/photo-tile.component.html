<div class="photo-tile card">
  <img class="photo"
       [src]="photo?.link"
       alt="photo">
</div>