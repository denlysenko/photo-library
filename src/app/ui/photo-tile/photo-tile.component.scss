.photo-tile {
  cursor: pointer;

  &.sm {
    width: 280px;
    height: 280px;
  }

  &.lg {
    width: 640px;
    height: 480px;
  }

  .photo {
    width: 100%;
    height: 100%;
  }
}
