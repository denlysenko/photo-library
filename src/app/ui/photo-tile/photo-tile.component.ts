import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

import { Photo } from '../../interfaces';

@Component({
  selector: 'app-photo-tile',
  templateUrl: './photo-tile.component.html',
  styleUrls: ['./photo-tile.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class PhotoTileComponent {
  @Input() photo: Photo;
  @Input() size: 'sm' | 'lg' = 'sm';
}
