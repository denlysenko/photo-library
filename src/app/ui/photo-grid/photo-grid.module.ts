import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { PhotoTileModule } from '../photo-tile/photo-tile.module';
import { PhotoGridComponent } from './photo-grid.component';

@NgModule({
  imports: [CommonModule, PhotoTileModule],
  declarations: [PhotoGridComponent],
  exports: [PhotoGridComponent],
})
export class PhotoGridModule {}
