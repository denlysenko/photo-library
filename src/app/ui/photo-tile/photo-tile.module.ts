import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { PhotoTileComponent } from './photo-tile.component';

@NgModule({
  imports: [CommonModule],
  declarations: [PhotoTileComponent],
  exports: [PhotoTileComponent],
})
export class PhotoTileModule {}
