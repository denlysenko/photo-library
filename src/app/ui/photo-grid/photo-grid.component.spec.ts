import { CommonModule } from '@angular/common';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { PhotoTileModule } from '../photo-tile/photo-tile.module';
import { PhotoGridComponent } from './photo-grid.component';

const photos = [
  {
    link: 'link',
    id: 'photoId',
  },
];

describe('PhotoGridComponent', () => {
  let component: PhotoGridComponent;
  let fixture: ComponentFixture<PhotoGridComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [CommonModule, PhotoTileModule],
      declarations: [PhotoGridComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(PhotoGridComponent);
    component = fixture.componentInstance;
    component.photos = photos;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(fixture).toMatchSnapshot();
  });

  describe('photoClicked', () => {
    it('should emit event with photo id', () => {
      const photoClickedSpy = jest.spyOn(component.photoClicked, 'emit');
      const photoTile = fixture.nativeElement.querySelector(
        '[data-testid=photo-tile]'
      );
      photoTile.click();
      expect(photoClickedSpy).toHaveBeenCalledWith(photos[0]);
    });
  });
});
