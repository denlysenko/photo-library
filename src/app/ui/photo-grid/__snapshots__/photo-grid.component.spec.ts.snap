// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoGridComponent should create 1`] = `
<app-photo-grid
  photoClicked={[Function EventEmitter_]}
  photos={[Function Array]}
>
  <div
    class="photo-grid"
  >
    
    <app-photo-tile
      data-testid="photo-tile"
    >
      <div
        class="photo-tile card sm"
      >
        <img
          alt="photo"
          class="photo"
          src="link"
        />
      </div>
    </app-photo-tile>
  </div>
</app-photo-grid>
`;
